### PR TITLE
Don't disable AWS src/dst checks in Calico IPv6

### PR DESCRIPTION
**Pull request by johngmyers**
/assign @olemarkus

### CODE REVIEW

**Comment by johngmyers** [comment]
Looks like I made it impossible to disable src/dst checks on IPv6 in the template. Should I support that or add validation against it?

**Comment by hakman** [comment]
> Looks like I made it impossible to disable src/dst checks on IPv6 in the template. Should I support that or add validation against it?

Let's not support it at all for IPv6. Should not be needed.

File: pkg/apis/kops/v1alpha2/networking.go [hunk 1]
```diff
@@ -108,8 +108,8 @@ type CalicoNetworkingSpec struct {
 	// AllowIPForwarding enable ip_forwarding setting within the container namespace.
 	// (default: false)
 	AllowIPForwarding bool `json:"allowIPForwarding,omitempty"`
-	// AWSSrcDstCheck enables/disables ENI source/destination checks (AWS only)
-	// Options: Disable (default), Enable, or DoNothing
+	// AWSSrcDstCheck enables/disables ENI source/destination checks (AWS IPv4 only)
+	// Options: Disable (default for IPv4), Enable, or DoNothing
```
Comment (hakman) [inline pkg/apis/kops/v1alpha2/networking.go:112]: Isn't the above `(AWS IPv4 only)` enough?
```suggestion
	// Options: Disable (default), Enable, or DoNothing
```

Comment (johngmyers) [reply pkg/apis/kops/v1alpha2/networking.go:112]: That could imply one has to override the default when on IPv6

Comment (hakman) [reply pkg/apis/kops/v1alpha2/networking.go:112]: To me the initial versions implies that there is some default for IPv4 and undefined for IPv6 and has to be set by the operator.
Would be nice to also add validation to not be able to set it for IPv6 at all.

Comment (johngmyers) [reply pkg/apis/kops/v1alpha2/networking.go:112]: This PR includes validation preventing it from being set to anything other than `DoNothing` on IPv6.

Comment (hakman) [reply pkg/apis/kops/v1alpha2/networking.go:112]: K, I'm good/

**Pull request merged**